### ISSUE TITLE
automated docs by fastapi not reachable

### ISSUE THREAD

**Issue by cowanAI** [gradio-app/gradio#5921]
### Describe the bug

<img width="313" alt="image" src="https://github.com/gradio-app/gradio/assets/128092052/6def05dc-426f-4d53-aef7-e295fd3e908c">

### Have you searched existing issues?  🔎

- [X] I have searched and found no existing issues

### Reproduction

Using this code:

ui.queue()
ui.launch(server_name="0.0.0.0", server_port=3000)
app =gr.mount_gradio_app(app, ui, path='/mypath') 

### System Info

```shell
latest
```

### Severity

Blocking usage of gradio

**Comment by abidlabs** [comment]
Hi @cowanAI could you please provide a complete example we can use to reproduce? And show the output you are seeing vs what you expect to see?

**Comment by cowanAI** [comment]
For sure:


from fastapi import FastAPI

import gradio as gr

import torch
import requests
from torchvision import transforms

model = torch.hub.load('pytorch/vision:v0.6.0', 'resnet18', pretrained=True).eval()
response = requests.get("https://git.io/JJkYN")
labels = response.text.split("\n")


app = FastAPI()

def predict(inp):
  inp = transforms.ToTensor()(inp).unsqueeze(0)
  with torch.no_grad():
    prediction = torch.nn.functional.softmax(model(inp)[0], dim=0)
    confidences = {labels[i]: float(prediction[i]) for i in range(1000)}    
  return confidences

demo = gr.Interface(fn=predict, 
             inputs=gr.inputs.Image(type="pil"),
             outputs=gr.outputs.Label(num_top_classes=3),
             examples=[["cheetah.jpeg"]],
             )
             
demo.launch()




app =gr.mount_gradio_app(app, demo, path='/mypath')

**Comment by cowanAI** [comment]
@abidlabs you will see that the /docs path is blocked that typically generates the API/swagger documentation automatically, also, I'm currently trying to integrate the gradio interface with a third party, and I built an ENDPOINT for the redirect URI using fast api called /callback, however I'm concerned if creating endpoints on fast api with this use case and redirect URIs is even allowed when gradio is on top of fast api. I would appreciate very much if you could clarify if this is possible.

**Comment by cowanAI** [comment]
![cheetah](https://github.com/gradio-app/gradio/assets/128092052/ce332388-d0df-42e7-95f4-b489bf4b4b30)

**Comment by cowanAI** [comment]
There is also the image you will need for that demo.

**Comment by cowanAI** [comment]
and this is the output when I try to get the fastapi swagger documentation 
<img width="224" alt="image" src="https://github.com/gradio-app/gradio/assets/128092052/58df6dbd-251d-443a-9f4c-b745a1973ac5">

**Comment by abidlabs** [comment]
Hi @cowanAI can you see if this is still an issue in Gradio 4.0? We've changed the way that routes are served and perhaps this is no longer an issue.

**Comment by abidlabs** [comment]
Closing for lack of followup -- let us know if we need to reopen.
